### DISCUSSION TITLE
Switched from cloud api integration to solaredge-modbus-multi integration

### DISCUSSION THREAD

**Discussion by peterlindeman** [WillCodeForCats/solaredge-modbus-multi#647] [Q&A] [answered]
Today I enabled modbus over tcp in my inverter and I have installed the latest release of this integration. So far so good. I get all necessary values in HA but I'm strugling with the Energy dashboard configuration.
For the cloud api integration I got the sensor sensor.solaredge_lifetime_energy which is configured in the energy dashboard for some years now so a lot of history I do not want to lose. With the modbus integration I will get the sensor sensor.solaredge_ac_energy_kwh which has the same info only in kWh where the cloud sensor is stored in Wh
When I replace the sensor in the energy dashboard configuration I lose my history.
What is the correct way to switch from the cloud integration to the modbus integration. Tried to find any information about it but cannot find it. Hopefully somebody knows the solution.

Thanks,
Peter

**Comment by WillCodeForCats** [comment]
You can change the units of the sensor, but internally it is native as watt-hours so there shouldn't be a difference in how the energy dashboard tracks it.

<img width="582" alt="Screenshot 2024-08-07 at 17 29 37" src="https://github.com/user-attachments/assets/d84a59de-e3e9-4c68-beab-9d194b5d3e6f">

**Comment by peterlindeman** [comment, marked as answer]
I have fixed it by deleting the old cloud API integration and after that renamed the new modbus sensor to the name the old sensor of the cloud API was using. It’s working now including al history 👍🏻
